### PR TITLE
Show preregistration (massVax) sites

### LINKED ISSUE #156: MassVax sites should appear on our site
Right now we are not showing the existence of the seven MassVax sites (Gilette, Fenway/Hynes, etc.) as there is no **immediate** availability via a _Sign Up_ button

If we flag these sites in the back-end, then we can show them to the user.

1. We will never display the (blue) "no appointments found" message because there is always something to show.
2. MassVax sites show up with ZIP Codes for proximity
3. Removes the (green) "preregistration alert" in favor of this.
4. Promotes **preregistration** to our users

**Here is the approved visual from the [#UX channel on Slack](https://macovidvaccines.slack.com/archives/C01N7C2G2KF/p1616105074009700):**

image

### PR DESCRIPTION
**Pull request by harcod**
Closes #156

Initial Draft of including preregistration buttons for the 7 MassVax sites. (Gilette, Fenway/Hynes, Dartmouth, etc)
There is a [back-end PR](https://github.com/livgust/covid-vaccine-scrapers/pull/161)  that need to be approved that is related to this issue.

**Here is a MassVax Location Card:**
image

**And new Filter UI**
image

To-do:
:heavy_check_mark: 1. update `appointmentData.service.js` to read flag provided by [back-end PR](https://github.com/livgust/covid-vaccine-scrapers/pull/161) (**This will be done by Issue #161** now)
:heavy_check_mark: 2. Add a cookie to save the setting of the checkbox.

### CODE REVIEW

**Comment by harcod** [comment]
Closes #159
If the "More Information" accordion only has a street address, then just show the address without the accordion.

**Comment by harcod** [comment]
Closes #136 
Cookies have been refactored into it's own file.  Now saves all "filter" settings in a cookie.  Also the expiration date for the cookie is 1 year (instead of the previous 1 day... oops)

**Comment by harcod** [comment]
I want to move this back to DRAFT status.

**Comment by harcod** [comment]
OK. This time for sure!

File: src/services/cookie.service.js [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+        encoded
+            .match(/.{1,2}/g)
+            .map((hex) => parseInt(hex, 16))
+            .map(applySaltToChar)
+            .map((charCode) => String.fromCharCode(charCode))
+            .join("");
+};
+
+// To create a cipher
+const cookieSalt = "XDpUY3ti0uEh8wUnWv6t";
```
Comment (tkjones117) [inline src/services/cookie.service.js:62]: I am curious how you came up with this salt?

File: src/services/cookie.service.js [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+const cookies = new Cookies();
+const cookieMaxAge = 365 * 86400;
+const cookieName = "macovidvaccines";
+
+let ourCookie = {};
+
+export function getCookie(key) {
+    const theCookie = cookies.get(cookieName, { doNotParse: true });
+    if (theCookie) {
```
Comment (tkjones117) [inline src/services/cookie.service.js:11]: could you come up with more descriptive names than `theCookie` and `ourCookie`?

**Comment by tkjones117** [comment]
Ran and this works, looks great.

**Comment by iann** [comment]
Something I'm noticing is that the checkbox state isn't being persisted in the cookie. Is this expected?

File: src/services/cookie.service.js (hunk 1 above)
Comment (harcod) [reply src/services/cookie.service.js:62]: random password generator

File: src/services/cookie.service.js (hunk 2 above)
Comment (harcod) [reply src/services/cookie.service.js:11]: Good idea! Changing to `encryptedCookie` and `plaintextCookie`.  These aren't exported, so the most important part about them is their encryption state.

**Comment by harcod** [comment]
> Something I'm noticing is that the checkbox state isn't being persisted in the cookie. Is this expected?

Not expected... it should be persisting **all of the filter settings** now. (The current production version only saves the 5-digit ZIP Code).  I just tried it locally (localhost:3000) and the cookie was being updated when clicking on _Update List_.  Do you have a test case?

**Comment by iann** [comment]
This is more of a Safari issue locally. Confirmed it works locally with Chrome.

**Review by iann — APPROVED** [review]
Works/looks good!

**Review by livgust — CHANGES_REQUESTED** [review]
Hey @harcod did you run this final version by UX? It's really annoying to show the mass vax sites by default because the sites with actual availability are peppered between them. I think if we show them, they should be at the bottom unless we're sorting by distance. But in any case, we should definitely run it by UX.

**Comment by harcod** [comment]
> Hey @harcod did you run this final version by UX? It's really annoying to show the mass vax sites by default because the sites with actual availability are peppered between them. I think if we show them, they should be at the bottom unless we're sorting by distance. But in any case, we should definitely run it by UX.

OK. I will get UX to sign off.

**Comment by harcod** [comment]
Closes #161 by checking for the massVax flag in the data file instead of URL matching **color** or **curative**

**Comment by harcod** [comment]
After much discussion in Slack, we decided to change this up quite a bit.

1. Make Preregistration look like a single location.
2. Buttons: [ Preregister ] [ Dismiss this reminder ]
3. Add more text to list all of the locations
4. Sort it.  **First** (I think it is important).  Once they preregister they can dismiss it.
5. The [ Dismiss this reminder ] button saves a cookie so that it never comes back.
6. Remove the "Show mass vaccination locations" in the filter panel

image

File: src/CovidAppointmentTable.js [hunk 3]
```diff
@@ -1,4 +1,5 @@
 /*eslint no-unused-vars: ["error", { "varsIgnorePattern": "[iI]gnored" }]*/
+/*eslint no-unused-vars: ["error", { "varsIgnorePattern": "[iI]gnored" }]*/
```
Comment (livgust) [inline src/CovidAppointmentTable.js:2]: is there a reason this is here twice?

File: src/services/cookie.service.js (hunk 1 above)
Comment (livgust) [reply src/services/cookie.service.js:70]: should this be in the code? should we move it to a secret in the repo or something?

**Review by livgust — CHANGES_REQUESTED** [review]
this looks awesome!! 2 quick comments

File: src/CovidAppointmentTable.js (hunk 3 above)
Comment (harcod) [reply src/CovidAppointmentTable.js]: removed.  I probably pressed Ctrl-D in my IDE to see a "diff" and got a "duplicate line" instead.

File: src/services/cookie.service.js (hunk 1 above)
Comment (harcod) [reply src/services/cookie.service.js]: moved to secrets

File: src/services/cookie.service.js (hunk 2 above)
Comment (harcod) [reply src/services/cookie.service.js]: done

**Review by livgust — APPROVED** [review]
Sorry for the delay!

**Pull request merged**